lib/bigint_pk.rb

```diff
@@ -48,27 +48,53 @@ def references(*args)
     end
   end
 
-  def install_patches!(adapter)
-    ca = ActiveRecord::ConnectionAdapters
+    module CompatibilityWithBigint
+      def self.included(base)
+        base.remove_possible_method :create_table
+      end
 
-    case adapter
-    when 'postgresql'
-      pk_module = PostgresBigintPrimaryKey
-      require 'active_record/connection_adapters/postgresql_adapter'
-      ca::PostgreSQLAdapter::NATIVE_DATABASE_TYPES[:primary_key] = 'bigserial primary key'
-    when /mysql\d+/
-      pk_module = MysqlBigintPrimaryKey
-      require 'active_record/connection_adapters/abstract_mysql_adapter'
-      ca::AbstractMysqlAdapter::NATIVE_DATABASE_TYPES[:primary_key] = 'bigint(20) auto_increment PRIMARY KEY'
-      ca::AbstractMysqlAdapter::NATIVE_DATABASE_TYPES[:integer] = { :name => "bigint", :limit => 6 }
-    else
-      raise "Only MySQL and PostgreSQL adapters are supported now. Tried to patch #{adapter}."
+      def create_table(table_name, options = {})
```

Comment:
For future reference https://github.com/rails/rails/blob/2fb658d1e39364926aebd09f264d395bd1863e56/activerecord/lib/active_record/migration/compatibility.rb#L29-L59


```diff
-          # Since 5.1 Postgres adapter uses bigserial type for primary
-          # keys by default and MySQL uses bigint. This compat layer makes old migrations utilize
-          # serial/int type instead -- the way it used to work before 5.1.
-          unless options.key?(:id)
-            options[:id] = :integer
-          end
-
-          if block_given?
-            super(table_name, options) do |t|
-              class << t
-                prepend TableDefinition
-              end
-              yield t
-            end
-          else
-            super
-          end
+          super
```